### ISSUE TITLE
Report of Reviewer 1

### ISSUE THREAD

**Issue by pneuvial** [computorg/published-202311-delattre-fim#2]
Associate Editor: *Pierre Neuvial*
Reviewer : (chose to remain anonymous)

## Reviewer: Reviewing history

- Paper submitted *2022-09-22*
- Reviews received *2022-01-15*
- Decision: major revision *2023-01-27*
- Paper revised *2023-07-10*
- Reviews received *2023-10-20*
- Paper accepted *2023-10-22*

## Summary

This paper addresses the computation of the Fisher information matrix (FIM) in statistical model with latent variable models. This paper contains essentially two parts:

- The first one is methodological and algorithmic and concerns the case when the observations are independent but possibly have different distributions. The authors first povide conditions for the consistency and asymptotic normality of the “sco” estimator of FIM. Then, they provide an expression of the “sco” estimator in the case of latent variable models. Finally, an algorithm is provided for the estimation of the FIM by using the output of a SAEM algorithm designed to compute the Maximum Likelihood estimator. The convergence of this algorithm is discussed: the authors provide a list of assumptions, some of them are very strong (see questions below) and these assumptions allow a direct application of a convergence result published in 1999. We can not say that Theorem 1 is a significant novelty (see question 6 below). Theorem 2 addresses the asymptotic normality of the novel FIM estimator; here again, the novelty is not significant, its proof relies essentially on published results and the use of the Delta-method.

- The second part is devoted to simulation studies. Two FIM estimators are compared through their bias and root mean squared deviation (first set of studies) or through relative bias and relative std (second set). A first set of studies uses two toy models, simulated data, and consider that the parameter is known; the limiting FIM is either known (first model) or estimated via Monte Carlo (second model). A second set of studies uses a more realistic and complex model (from pharmacokinetics); the limiting value is not explicit and is not estimated. There are many details on the models at hand, on how the experiments are obtained. There are a lot of numerical investigations, really interesting; including a comparison to the literature.

### Conclusion

My conclusion is that the content of this paper is interesting and can be considered for publication in COMPUTO after a major revision. This revision essentially concerns the theoretical and methodological part; more rigorous statements are expected (or at least, detailed references to the literature) as well as a presentation of the results which clearly identify the novelty of the theoretical results and of the algorithmic contribution: detailed questions and remarks are below.

The codes do not seem to be provided.

1. Proof of Proposition 1: please provide a reference for the SLLN, and check all the assumptions; similarly, for the CLT, provide a reference for a detailed proof (e.g. explaining the Taylor expansion around zero for a complex-valued function).

> To be consistent with the remark on asymptotic properties of the FIM estimates and also to keep focus on the main contribution of the paper, we replace Proposition 1 where several assumptions were missing by a remark indicating that asymptotic properties also hold in the independent non identically distributed setting under stronger assumptions. We precise references for assumptions and these related results in the revised version.

2. Proposition 2. A detailed proof or more detailed references should be provided - for example, explaining how an integrability condition on the derivative of $\log g_i$ implies that $\log f$ is differentiable and integrable.

> We thank the reviewer for this comment. Indeed our statement was incomplete. We have corrected this by adding assumptions on $f$ and $\log f$ and some additional reference on the classical Fisher identity.

3. Section 3.2.1. and after: it is not correct to define $\hat{\Delta}_i$ as a function of the variable $\hat{\theta}(s)$ only. It also depends on $s_i$. This modification has to be propagated through the paper carefully.

> We thank again the reviewer for this comment. We have corrected this all along the paper.

4. Theorem 1 assumes that with probabillity one, the closure of the SAEM sequence is a compact set. This stability condition is very difficult to check in practice. To my knowledge, it is satisfied by using a projection step; in that case, the convergence analysis of the stochastic approximation scheme is not covered by the Delyon et al. (1999) paper, and is not covered by Theorem 1. Please comment how to check this assumption in practice in such a way that the Theorem 1 still applies.

We admit that this stability condition of the previous Theorem 1 is very difficult to check in practice. However, we emphasize that the proposed algorithm can be very useful in practice. We therefore choose to postpone it, as well as the corresponding theoretical results in the Appendix of the revised version. Following the reviewer’s suggestion, we provide in the revised version a new algorithm including a projection step at each iteration through truncation on random boundaries. We choose to follow the lines of the contribution of Andrieu et al (2005) since it stated conditions that can be verified easily in practice. We thus provide an algorithm and a theoretical result that are more widely applicable since i) the simulation step of the algorithm can be performed using a transition kernel of a Markov chain rather than simulation from the posterior distribution; ii) conditions to check for the theoretical result can be easily verified in practice. We emphasize that the algorithm is proposed using some general transition kernel of a Markov chain whereas the theoretical result is established for the transition kernel being this of the random walk Metropolis Hastings algorithm. Note that this setting is very common in practice. We also mention how our result can be adapted to a more general transition kernel based on the result of Andrieu et al (2005).

5. Theorem 1: what is $\ell$, the (observed) likelihood used in the definition of since the observations are not iid: is it the joint likelihood of the observations ? What is $I(\theta)$ (again, the observations are not iid) ? Does it exist a mathematical expression of the possible $I(\theta)$ and the $\hat{\Delta}_i(\hat{\theta}(s))$ ?

> We thank again the reviewer for this comment. We have added all the mathematical expressions of these quantities in the revised version.

6. As far as I understand, both Theorem 1 and Theorem 2 consist in (a) showing that the SAEM sequence or its averaged version converges almost-surely to a set; this is done by applying known results (there is no novelty in this step) and then (b) claiming that since the estimator is assumed to be a continuous function of $s$, then it converges to the estimator evaluated on the limit set of $s$-sequence. If my understanding is correct, section 3.2.2. should be written to clearly outline the novelty of the theoretical analysis and remove all the assumptions which are provided in the appendix section; otherwise, please comment where the originality is w.r.t. to the proof of Delyon et al (1999).

> Please note that in the revised version, following a previous suggestion of Reviewer 1, we provide now an algorithm with a projection step; therefore we have postponed in Appendix previous Theorem 1 and Theorem 2 of the initial version. However we have highlighted the novelty of the theoretical analysis, in particular getting the FIM estimate as a by-product of the MLE with the proposed stochastic algorithm requires to assume inde- pendent vectors $(Y_i , Z_i )_{1 \leq i \leq n}$ and to evaluate individual sufficient statistic estimates.

7. Please, comment the interest of a CLT for the FIM estimate. Is the asymptotic covariance computable in practice ? Does Theorem 2 still hold when the step size used when defining the original sequence $s^k$ is $1/(k+1)?

> The CLT for the FIM estimate could allow for example to compute asymptotic confidence interval in post-model selection inference using plug-in (Charkhi and Claeskens, 2018). However, we agree with the reviewer's comment that the asymptotic covariance is not often easy to compute in practice. Besides we detailed in the revised version the condition on the step size sequence to get the CLT which was quite imprecise in the first version. Indeed Theorem 2 does not hold for the step-size sequence $1/(k+1)$ (see Delyon et al 1999 for more details and assumption (SAEM1’) on the step-size sequence).

8. In Section 3.2.1, the estimator is the mean value over the observations, of the derivative of the complete log-likelihood, evaluated at the current value of the statistic $s^k$ and of the parameter $\theta^k$. In Section 3.2.2., the estimator does not follow the same strategy: it is the mean value over the observations of the $\Delta_i^K$ – which is the *cumulated* averaged sum of the derivative of the complete log-likelihood evaluated at the current parameter. These two strategies do not look similar: for example, if the complete data model is from the curve exponential family as in Section 3.2.1., the algorithms do not look equivalent. Please comment this point.

> We emphasize here the difference between both algorithms proposed in Sections 3.2.1 and 3.2.3. The first one is proposed in models belonging to the curved exponential family. In such models, the estimation of parameters and of expected individual sufficient statistics ($s_i$) can be performed in parallel simultaneously during the algorithm. Note also that in those models the quantities ($\Delta_i$) depend only on the estimates ($s_i$) as highlighted in question 3 of Reviewer 2. Therefore the FIM estimate can be evaluated as an explicit function of all these quantities. Conversely, the second algorithm is proposed for more general models potentially out of the curved exponential family. In this case, it is not possible to compute in parallel both estimates of parameters and expected sufficient statistics since we do not assume that each individual complete likelihood belongs to the curved exponential family. Therefore we compute direct estimates of the quantities ($\Delta_i$) during the algorithm, in the same spirit of the approximation of the expected complete log-likelihood performed with the sequence ($Q_k$). Note that as the update of the quantity $Q_k$ in Section 3.2.3 is strictly equivalent to that of the quantity ($s^k_i$) in the previous section 3.2.1, this is not the case for the quantity $∆^k_i$ since it depends explicitly on the current value of the parameter $\theta_{k−1}$ which is plugged-in. The advantage of this second algorithm is that it can be applied in a wide range of models, also those which do not belong to the curved exponential family.

9. Section 4, the simulation study: the Poisson model is an iid case isn’t it ? what about the linear mixed effect model ? please provide the expression of $\phi_i$, $s_i(Z)$, $\psi_i \hat{\theta}$. Section 4.1.2. the parameter $\theta^*$ is known but it is not clear how the FIM estimators are computed. Do they rely on a SA algorithm for the computation of the incomplete data likelihood ? please explain how the FIM estimators are computed.

> You are right. In the Poisson model, the variables $Y_{i}$ are iid random variables. In the linear mixed-effects model, the variables $Y_{i}$ composed of the individual observations $\left(y_{i j}\right)_{1 \leq j \leq J}$ are also iid random variables. Both models belong to the curved exponential family. In the linear mixed-effects model, we have

$$
S_{i}\left(z_{i}\right)=\left(\begin{array}{c}
z_{i}^{2} \\
z_{i}
\end{array}\right), \psi_{i}(\theta)=-\frac{J}{2} \log \left(\sigma^{2}\right)-\frac{1}{2} \log \left(\eta^{2}\right)-\frac{1}{2} \sum_{j=1}^{J}\left(y_{i j}-\beta\right)^{2},$$

$$\phi_{i}(\theta)=\left(\begin{array}{c}
-\frac{1}{2 \eta^{2}} - \frac{J}{2 \sigma^{2}} \\
-2\sum_{1\leq j \leq J} (y_{i j}-\beta)
\end{array}\right),$$

10. Section 4.1.2. the parameter $\theta^*$ is known but it is not clear how the FIM estimators are computed. Do they rely on a SA algorithm for the computation of the incomplete data likelihood ? please explain how the FIM estimators are computed.

> and in the Poisson mixture model, we have

$$
\begin{gathered}
S_{i}\left(z_{i}\right)=\left(\begin{array}{c}
1_{z_{i}=1} \\
\vdots \\
1_{z_{i}=K}
\end{array}\right), \psi_{i}(\theta)=-\log \left(y_{i} !\right), \\
\phi_{i}(\theta)=\left(\begin{array}{c}
-\lambda_{1}+y_{i} \log \left(\lambda_{1}\right)+\log \left(\alpha_{1}\right) \\
\vdots \\
-\lambda_{K-1}+y_{i} \log \left(\lambda_{K-1}\right)+\log \left(\alpha_{K-1}\right) \\
-\lambda_{K}+y_{i} \log \left(\lambda_{K}\right)+\log \left(1-\sum_{1 \leq k \leq K} \alpha_{k}\right)
\end{array}\right) .
\end{gathered}
$$

However, this decomposition and thus the algorithm described in Section 3 are not helpful to compute the Fisher information matrix estimates $I_{n,sco}$ and $I_{n,obs}$ which are explicit by following formula 3 and Louis’ formula (Louis, 1982) in the Poisson mixture model. This is now clearly mentioned in Section 4.1. On the contrary, implementing our algorithm in the model studied in section 4.2.1 relies on the knowledge of statistics $S_i(z_i)$, $\psi_i(\theta)$, and $\phi_i(\theta)$.

Their expressions have been added to the text.
